src/quickstart.ts

```diff
@@ -1,149 +1,163 @@
-    return fs.writeFile(this.cmakeFilePath, init);
-  }
+  await fs.writeFile(sourceCodeFilePath, [
+    '#include <iostream>',
+    '',
+    'void say_hello(){',
+    `    std::cout << "Hello, from ${projectInformation.name}!\\n";`,
+    '}',
+    '',
+  ].join('\n'));
+  return sourceCodeFilePath;
+}
 
-  private async createLibrarySourceFile() {
-    return fs.writeFile(this.sourceCodeFilePath, [
-      '#include <iostream>',
-      '',
-      'void say_hello(){',
-      `    std::cout << "Hello, from ${this.projectName}!\\n";`,
-      '}',
-      '',
-    ].join('\n'));
+async function createMainSourceCodeFile(projectInformation: CreateProjectInformation) {
+  const sourceCodeFilePath = path.join(projectInformation.sourceDirectory, 'main.cpp');
+  if (await fs.exists(sourceCodeFilePath)) {
+    throw new Error('Main source file is already present.');
   }
 
-  private async createMainSourceCodeFile() {
-    return fs.writeFile(this.sourceCodeFilePath, [
-      '#include <iostream>',
-      '',
-      'int main(int, char**) {',
-      '   std::cout << "Hello, world!\\n";',
-      '}',
-      '',
-    ].join('\n'));
-  }
+  await fs.writeFile(sourceCodeFilePath, [
+    '#include <iostream>',
+    '',
+    'int main(int, char**) {',
+    '   std::cout << "Hello, world!\\n";',
+    '}',
+    '',
+  ].join('\n'));
+
+  return sourceCodeFilePath;
+}
 
-  private async createExampleSourcecodeFile() {
-    if (!(await fs.exists(this.sourceCodeFilePath))) {
-      switch (this.type) {
-      case ProjectType.Library:
-        return this.createLibrarySourceFile();
-      case ProjectType.Exectable:
-        return this.createMainSourceCodeFile();
-      }
+export async function createProject(projectInformation: CreateProjectInformation):
+    Promise<GeneratedProjectFiles> {
+  const cmakeFile = await createCMakeListFile(projectInformation);
+
+  let sourceFile: string = '';
+  try {
+    if (projectInformation.type == ProjectType.Exectable) {
+      sourceFile = await createMainSourceCodeFile(projectInformation);
+    } else if (projectInformation.type == ProjectType.Library) {
+      sourceFile = await createLibrarySourceFile(projectInformation);
     }
-  }
+  } catch (err) { log.debug(err); }
 
-  public async createProject() {
-    await this.createCMakeListFile();
-    await this.createExampleSourcecodeFile();
-  }
+  return {cmakeListFile: cmakeFile, sourceFiles: [sourceFile]} as GeneratedProjectFiles;
 }
 
-export interface QuickStartCallbacks {
+export interface UiControlCallbacks {
   onProjectNameRequest: () => Promise<string|undefined>;
-  onProjectTypeRequest: (items: ProjectTypeDesciptor[]) => Promise<ProjectTypeDesciptor|undefined>;
-  onOpenSourceFiles: (filePaths: string) => void;
+  onProjectTypeRequest: (items: ProjectTypeDesciption[]) => Promise<ProjectTypeDesciption|undefined>;
+  onOpenSourceFiles: (filePaths: GeneratedProjectFiles) => Promise<void>;
   onError: (message: string) => void;
+  onWarning: (message: string) => void;
```

Comment:
These are a bit too close to event names, even though they are not.

Change `onProjectNameRequest` to `getProjectName`, and so on.